### PR TITLE
virts-3151 - API endpoint documentation (create, update, delete agents + retrieve deploy commands)

### PR DESCRIPTION
**Pull request by ArtificialErmine**
## Description

Documents V2 API endpoints for creating, updating, and deleting agents and retrieving deploy commands

## Type of change

- [X] This change requires a documentation update

## How Has This Been Tested?

This has been run and the produced Swagger docs examined.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [NA] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/api/v2/handlers/agent_api.py [hunk 1]
```diff
@@ -59,26 +59,32 @@ async def update_agent(self, request: web.Request):
         agent = await self.update_object(request)
         return web.json_response(agent.display)
 
-    @aiohttp_apispec.docs(tags=['agents'])
+    @aiohttp_apispec.docs(tags=['agents'], summary="Update an agent, or if a existing agent match cannot be found, "
+                                                   "create one. Use the paw field in the URL to specify matching "
+                                                   "criteria and the fields from the AgentSchema in the request body"
+                                                   " to specify new field values.")
```
Comment (bleepbop) [inline app/api/v2/handlers/agent_api.py:65]: For the endpoints that have url parameters, you can add something like the following to the `aiohttp_apispec.docs` decorator to describe the parameters:
```
parameters=[{
  'in': 'path',
  'name': 'paw',
  'schema': {'type': 'string'},
  'required': 'true',
  'description': 'paw of the Agent to be retrieved.'
}],
```

File: app/api/v2/handlers/agent_api.py [hunk 2]
```diff
@@ -59,26 +59,32 @@ async def update_agent(self, request: web.Request):
...
     @aiohttp_apispec.request_schema(AgentSchema(partial=True))
     @aiohttp_apispec.response_schema(AgentSchema)
     async def create_or_update_agent(self, request: web.Request):
         agent = await self.create_or_update_object(request)
         return web.json_response(agent.display)
 
-    @aiohttp_apispec.docs(tags=['agents'])
+    @aiohttp_apispec.docs(tags=['agents'], summary="Delete an agent. Use the paw field in the URL to specify matching "
+                                                   "criteria for the agent(s) to delete.")
     @aiohttp_apispec.response_schema(AgentSchema)
```
Comment (bleepbop) [inline app/api/v2/handlers/agent_api.py:74]: Can you add a `description` to the `@aiohttp_apispec.response_schema` for each of these endpoints?

File: app/api/v2/handlers/agent_api.py [hunk 3]
```diff
@@ -59,26 +59,32 @@ async def update_agent(self, request: web.Request):
         agent = await self.update_object(request)
         return web.json_response(agent.display)
 
-    @aiohttp_apispec.docs(tags=['agents'])
+    @aiohttp_apispec.docs(tags=['agents'], summary="Update an agent, or if a existing agent match cannot be found, "
```
Comment (bleepbop) [inline app/api/v2/handlers/agent_api.py:62]: The `summary` key is essentially a title for these endpoints. For each endpoint, can you add a `summary` and `description` for the `@aiohttp_apispec.docs`?

I think for these endpoints, what is currently the `summary` is a bit more like the `description` (i.e. a more detailed writeup about the endpoint), and the `summary` can just be something like "Replace an Agent."

Comment (ArtificialErmine) [reply app/api/v2/handlers/agent_api.py]: Got it - didn't know description was a field.

File: app/api/v2/handlers/agent_api.py (hunk 2 above)
Comment (ArtificialErmine) [reply app/api/v2/handlers/agent_api.py:92]: Yep, will do.

File: app/api/v2/handlers/agent_api.py (hunk 1 above)
Comment (ArtificialErmine) [reply app/api/v2/handlers/agent_api.py]: I think that makes sense. I've added a version of this for every url-based parameter in these 6 endpoint entries.

File: app/api/v2/handlers/agent_api.py (hunk 2 above)
Comment (bleepbop) [reply app/api/v2/handlers/agent_api.py:92]: I think this may have accidentally been left out of the last commit

**Review by bleepbop — CHANGES_REQUESTED** [review]
Just needs `descriptions` in the `@aiohttp_apispec.response_schema` decorators for each endpoint, and will be all set after that

File: app/api/v2/handlers/agent_api.py (hunk 2 above)
Comment (ArtificialErmine) [reply app/api/v2/handlers/agent_api.py]: Yep, my bad. Should be fixed now.

**Review by bleepbop — APPROVED** [review]

**Review by yee-jonathan — APPROVED** [review]
Looks good to go. Decorators all have documentations

**Pull request merged**